swiftsimio/reader.py

```diff
@@ -407,7 +407,15 @@ def postprocess_header(self):
 
         for field, name in metadata.metadata_fields.header_unpack_string.items():
             try:
-                setattr(self, name, self.header[field].decode("utf-8"))
+                # Deal with h5py's quirkiness that fixed-sized and variable-sized
+                # strings are read as strings or bytes
+                # See: https://github.com/h5py/h5py/issues/2172
+                raw = self.header[field]
+                try:
+                    string = raw.decode("utf-8")
+                except:
```

Comment:
Probably best not to use a bare except. Apparently that even catches ctrl-c! Maybe we can just catch the AttributeError when it has no decode() method, or use isinstance to see if we got str or bytes?

Reply:
Ah! I learned something about exceptions. :)

Thanks!